Source/FieldSolver/SpectralSolver/SpectralAlgorithms/AvgGalileanAlgorithm.cpp

```diff
@@ -0,0 +1,371 @@
+    X3_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    X4_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Theta2_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+
+    A1_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    A2_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+
+    Rhoold_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Rhonew_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+    Jcoef_coef = SpectralComplexCoefficients(ba, dm, 1, 0);
+
+    // Fill them with the right values:
+    // Loop over boxes and allocate the corresponding coefficients
+    // for each box owned by the local MPI proc
+    for (MFIter mfi(ba, dm); mfi.isValid(); ++mfi){
+
+        const Box& bx = ba[mfi];
+
+        // Extract pointers for the k vectors
+        const Real* modified_kx = modified_kx_vec[mfi].dataPtr();
+#if (AMREX_SPACEDIM==3)
+        const Real* modified_ky = modified_ky_vec[mfi].dataPtr();
+#endif
+        const Real* modified_kz = modified_kz_vec[mfi].dataPtr();
+        // Extract arrays for the coefficients
+        Array4<Real> C = C_coef[mfi].array();
+        Array4<Real> S_ck = S_ck_coef[mfi].array();
+        Array4<Real> C1 = C1_coef[mfi].array();
+        Array4<Real> S1 = S1_coef[mfi].array();
+        Array4<Real> C3 = C3_coef[mfi].array();
+        Array4<Real> S3 = S3_coef[mfi].array();
+
+        Array4<Complex> Psi1 = Psi1_coef[mfi].array();
+        Array4<Complex> Psi2 = Psi2_coef[mfi].array();
+        Array4<Complex> Psi3 = Psi3_coef[mfi].array();
+        Array4<Complex> X1 = X1_coef[mfi].array();
+        Array4<Complex> X2 = X2_coef[mfi].array();
+        Array4<Complex> X3 = X3_coef[mfi].array();
+        Array4<Complex> X4 = X4_coef[mfi].array();
+        Array4<Complex> Theta2 = Theta2_coef[mfi].array();
+        Array4<Complex> A1 = A1_coef[mfi].array();
+        Array4<Complex> A2 = A2_coef[mfi].array();
+
+        Array4<Complex> CRhoold = Rhoold_coef[mfi].array();
+        Array4<Complex> CRhonew = Rhonew_coef[mfi].array();
+        Array4<Complex> Jcoef   = Jcoef_coef[mfi].array();
+        // Extract reals (for portability on GPU)
+
+        Real vx = v_galilean[0];
+        Real vy = v_galilean[1];
+        Real vz = v_galilean[2];
+
+        // Loop over indices within one box
+        ParallelFor(bx,
+        [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+        {
+            // Calculate norm of vector
+            const Real k_norm = std::sqrt(
+                std::pow(modified_kx[i], 2) +
+#if (AMREX_SPACEDIM==3)
+                std::pow(modified_ky[j], 2) +
+                std::pow(modified_kz[k], 2));
+#else
+                std::pow(modified_kz[j], 2));
+#endif
+
+            // Calculate coefficients
+            constexpr Real c = PhysConst::c;
+            constexpr Real c2 = PhysConst::c*PhysConst::c;
+            constexpr Real ep0 = PhysConst::ep0;
+            const Complex I{0.,1.};
+            if (k_norm != 0){
```

Comment:
Here I think it would be better to add the suffix `_rt` for single precision runs:
```suggestion

            if (k_norm != 0._rt){
```